benchmark.go

```diff
@@ -187,13 +223,74 @@ func runBenchmark(out output, logger *zap.Logger, allOpts Options, resolved reso
 		zap.Int("totalRequests", overall.totalRequests),
 		zap.Time("startTime", start),
 	)
-
+	// Print out errors
+	// TODO: allow errors to be printed in JSON format for output consistency
 	overall.printErrors(out)
-	overall.printLatencies(out)
 
-	out.Printf("Elapsed time:      %v\n", (total / time.Millisecond * time.Millisecond))
-	out.Printf("Total requests:    %v\n", overall.totalRequests)
-	out.Printf("RPS:               %.2f\n", float64(overall.totalRequests)/total.Seconds())
+	latencyValues := overall.getLatencies(out)
+
+	summary := Summary{
+		ElapsedTime:   (total / time.Millisecond * time.Millisecond).String(),
+		TotalRequests: overall.totalRequests,
+		RPS:           float64(overall.totalRequests) / total.Seconds(),
+	}
+
+	if opts.Format == "json" || opts.Format == "JSON" {
```

Comment:
shouldn't have to check the Format multiple times, maybe just cache the result as a boolean `formatAsJson` when you did the check above and use it here